server/services/withings/lib/withings.init.js

```diff
@@ -0,0 +1,358 @@
+          max: 100000,
+          feature_state: [],
+        };
+      }
+
+      // Build a feature state
+      const uniqueSateId = uuid.v4();
+      const createDate = new Date(currentGroup.created * 1000);
+      const featureState = {
+        id: uniqueSateId,
+        device_feature_id: tmpFeature.id,
+        value: element.value * 10 ** element.unit,
+        // created_at: `${createDate.getFullYear()}-${createDate.getMonth() + 1}-${createDate.getDate()}
+        // ${createDate.getHours()}:${createDate.getMinutes()}:${createDate.getSeconds()}`,
+        created_at: createDate,
+        updated_at: new Date(),
+      };
+
+      tmpFeature.feature_state.push(featureState);
+
+      if (!tmpFeature.last_value || tmpFeature.last_value_changed < createDate) {
+        tmpFeature.last_value_changed = createDate;
+        tmpFeature.last_value = featureState.value;
+        tmpFeature.last_value_string = `${featureState.value}`;
+      }
+
+      if (isNewFeature) {
+        features.push(tmpFeature);
+      }
+    });
+  }
+
+  return features;
+}
+
+/**
+ * @description Build and save withings device and init feature values.
+ * @param {string} userId - Gladys userId of current session.
+ * @returns {Promise} Resolve with withings device added.
+ * @example
+ * withings.init('b2f2c27f0bf3414e0fe3facfba7be9455109409a');
+ */
+async function init(userId) {
+  const { serviceId } = this;
+  const oauth2Manager = new OAuth2Manager(this.gladys);
+
+  const userResult = await oauth2Manager.executeQuery(
+    serviceId,
+    userId,
+    this.integrationName,
+    'get',
+    `${this.withingsUrl}/v2/user`,
+    'action=getdevice',
+  );
+
+  const devices = [];
+  const mapOfDeviceByWithingsDeviceId = new Map();
+  await userResult.data.body.devices.forEach((element) => {
+    if (element) {
+      // Build one gladys device for each withings device found
+      const newDevice = buildNewDevice(element, serviceId);
+      devices.push(newDevice);
+      mapOfDeviceByWithingsDeviceId.set(element.deviceid, newDevice);
+    }
+  });
+
+  const measureResult = await oauth2Manager.executeQuery(
+    serviceId,
+    userId,
+    this.integrationName,
+    'get',
+    `${this.withingsUrl}/measure`,
+    'action=getmeas',
+  );
+
+  // logger.warn(measureResult.data.body);
+  const mapOfMeasuresGrpsByWithingsDeviceId = new Map();
+  await measureResult.data.body.measuregrps.forEach((element) => {
+    if (element) {
+      // Build map of measuregrps by withings device id
+      let measureList = mapOfMeasuresGrpsByWithingsDeviceId.get(element.deviceid);
+      if (!measureList) {
+        measureList = [];
+      }
+      measureList.push(element);
+      mapOfMeasuresGrpsByWithingsDeviceId.set(element.deviceid, measureList);
+    }
+  });
+
+  const mapOfFeatureByWithingsDeviceId = new Map();
+  await mapOfMeasuresGrpsByWithingsDeviceId.forEach(function buildFeatureByGrps(value, key) {
+    value.forEach(function parseMeasureGroups(currentGroup) {
```

Comment:
Could it be replaced by an arrow function too?

Reply:
i transform to arrow function